packages/fxa-auth-server/lib/oauth/db/accessToken.js

```diff
@@ -87,31 +89,92 @@ class AccessToken {
       createdAt: this.createdAt.getTime(),
       profileChangedAt: this.profileChangedAt,
       expiresAt: this.expiresAt.getTime(),
-      token: this.token.toString('hex'),
     };
   }
 
+  static generate(
+    clientId,
+    name,
+    canGrant,
+    publicClient,
+    userId,
+    email,
+    scope,
+    profileChangedAt,
+    expiresAt,
+    ttl
+  ) {
+    const token = unique.token();
+    const tokenId = encrypt.hash(token);
+    return new AccessToken(
+      tokenId,
+      clientId,
+      name,
+      canGrant,
+      publicClient,
+      userId,
+      email,
+      scope,
+      // Truncated createdAt to the second to match mysql
+      new Date(new Date().setMilliseconds(0)),
+      profileChangedAt || 0,
+      expiresAt || new Date(Date.now() + (ttl * 1000 || MAX_TTL)),
+      // This is the one and only time the caller can get at the unhashed token.
+      token
```

Comment:
Not that it matters here and now, but I half wonder, in a more strongly typed future if we should have a `FreshAccessToken` subtype with the token while the base class doesn't.

Reply:
I was inches away from returning a two-tuple of `(raw token bytes, access token object)` from this function instead, for similar reasons.